blt/src/Commands/HumsciCommand.php

```diff
@@ -241,4 +243,36 @@ protected function updateSite($multisite) {
     $this->say("Finished deploying updates to $multisite.");
   }
 
+  /**
+   * Update autoloader in composer.json.
+   *
+   * To allow us the ability to create TestBase.php files that can be inherited
+   * by test classes, we have to specify each namespace with the appropriate
+   * directory for the test files. Since composer doesn't do this dynamically,
+   * we have to manually build the autoloader data with all available tests.
+   *
+   * @command update-autoloader
+   */
+  public function updateAutoloader() {
+    $root = $this->getConfigValue('repo.root');
+    $humsci_modules = $this->getConfigValue('docroot') . '/modules/humsci';
+
+    $classes = [];
+    foreach ($this->rglob("$humsci_modules/*Test.php") as $path) {
+      $relative_path = str_replace("$root/", '', $path);
+
+      var_dump($relative_path);
+
+      $module_path = substr($relative_path, 0, strpos($relative_path, '/tests/'));
+      $module = substr($module_path, strrpos($module_path, '/') + 1);
+
+      $classes["Drupal\\Tests\\$module\\"] = "$module_path/tests";
+    }
+
+    $composer = json_decode(file_get_contents("$root/composer.json"), TRUE);
+    $composer['autoload-dev']['psr-4'] = $classes;
```

Comment:
re-writing the composer.json file seems dangerous. Is there no way to hook in to the creation of the composer.lock or the creation of the autoloader.php files?

Reply:
from what i've been able to tell, the composer.lock doesnt do much for the autoloader. if you view the composer.lock in this PR, theres no reference to `hs_basic_page` or any of the other psr-4 spaces i've defined form the composer.json. https://getcomposer.org/doc/01-basic-usage.md#autoloading
The composer.json is the only way unless i wan to add a `post-autoload-dump` composer script and modify the autoload file directly.